### DISCUSSION TITLE
Livewire encountered corrupt data when trying to hydrate [foo] component

### DISCUSSION THREAD

**Discussion by chatisk** [livewire/livewire#5597] [1. Help]
Greetings everyone,

After an extended research accross all discussion regarding this tough issue, I want to open this discussion in case someone has any clue on fixing this. My system is deployed on AWS Lambda with Laravel Vapor. Thing is that i can't replicate locally this issue. 

I have 3 components that this occurs.

The 1 is a nested component, so I guess it might be tackled with a wire:key. The rest 2, are full page components.
One of these 2, executes this on `render()` method.
`
$options = $this->fetchProductOptions()->groupBy('product_id')->paginate(10);
`
`
return view('livewire.embedded.by-date', ['allOptions' => $options])
`

plus it has 4 custom query string + pagination.

As per @joshhanley 's suggestions, One of the reasons that are triggering this error is when the ordering of an array changes via JS, meaning that groupBy and re-order by JS are mis-matching. But I read on another thread that there is mismatch between the query string types. https://github.com/livewire/livewire/discussions/5352 . 

The most weird of all things, is that I can't replicate the errors locally, neither on production. They happen once in a while, but still annoying for the user to experience. 

Last i want to add is that the hashCode when the last error occured was "00000000" . Smth i have never encountered in any of the discussions mentioning this error.

**Comment by chatisk** [comment]
More code. This is for the groupBy issue perhaps. Can't reproduce

```php
<?php


class ByDate extends Component
{
    use WithPagination;
    use WithEmbeddedByDate;
    use CleanReservationOnClose;

    /**
     * The start date for our search
     *
     * @var string
     */
    public string $date = '';

    /**
     * 
     *
     * @var string
     */
    public string $currency = 'EUR';

    /**
     * 
     *
     * @var string
     */
    public string $c = '';

    /**
     * 
     *
     * @var string
     */
    public string $g = '';

    /**
     * 
     *
     * @var string
     */
    public string $r = '';

    /**
     * 
     *
     * @var string
     */
    public string $p = '';

    public ?Collection $options;

    public ?Collection $availableOptions;

    public ?Collection $resourcesOptions;

    public ?Collection $resourcesAvailableOptions;

    public $listeners = [
        'date:selected' => 'dateSelected'
    ];

    protected $queryString = ['date', 'c', 'g', 'p', 'currency', 'r'];

    /**
     * @return Factory|View|Application
     */
    public function render(): Factory|View|Application
    {
        $options = $this->fetchProductOptions()->groupBy('product_id')->paginate(10);
        return view('livewire.embedded.by-date', ['allOptions' => $options])
            ->layout('layouts.embedded');
    }

    public function mount()
    {
        if (!$this->date) {
            $this->date = Carbon::now()->format('Y-m-d');
        }
        $this->options                          = $this->fetchProductOptions()->groupBy('product_id')->get();
        $this->availableOptions                 = $this->fetchProductOptions()->get();
        if ($this->r) {
            $this->resourcesOptions             = $this->fetchProductOptions(true, $this->options->pluck('product_id')->toArray())->groupBy('product_id')->get();
            $this->resourcesAvailableOptions    = $this->fetchProductOptions(true)->get();
        }
    }
}

```

**Comment by chatisk** [comment]
Tried the back-button-cache, had some effect. will test it in time and update here.

**Reply by joshhanley** [reply]
@chatisk yeah back button cache could cause this issue. Also using the `groupBy()` is likely causing issues.

**Reply by chatisk** [reply]
Well , I was getting this error with the back button. As soon as I put the cache it stopped doing. I wonder if it is the correct solution. Thanks for taking the time to reply

**Reply by joshhanley** [reply]
@chatisk sometimes, if it works, then it's the correct solution 😆 especially with this corruption issue!

**Comment by d-shannon** [comment]
Hey @joshhanley,

Do you think it would be technically possible for V3 to provide more detailed info on where the corruptions originate? I have same issue in prod, can't track it down at all. Unfortunately I can't replicate locally. Would help massively in resolving these issues!

**Reply by ChrisSantiago82** [reply]
It also be that the data you are passing around is being changed by JavaScript.
Once I had in a value of -0 in php. The error happened very randomly to me.
-0 was fine in php, but then was changed to 0 in JavaScript

**Reply by joshhanley** [reply]
@d-shannon yep!

Typically the corruptions happen due to incompatible datatypes between javascript and php, such as the -0 suggested by @ChrisSantiago82.

We think we've managed to work out a solution for V3 where this is no longer an issue 🤞so hopefully in future when you get this error, it will actually just be because someone is trying to tamper with the data.

But we also looking at adding a feature to make debugging easier.

Hope this helps!
